### PR TITLE
Feature request:  Allow us to change the PID tuning method with M303

### PR DESCRIPTION
**Pull request by madhunm**
This patch allows us to change the the PID tuning method without reflashing firmware.

### CODE REVIEW

File: Marlin/temperature.cpp [hunk 1]
```diff
@@ -86,6 +86,7 @@ uint8_t Temperature::soft_pwm_bed;
       float Temperature::Kc = DEFAULT_Kc;
     #endif
   #endif
+  static int pid_method = 0;
```
Comment (Blue-Marlin) [inline Marlin/temperature.cpp:89]: 'static' here means - don't make this variable accessible from other modules.

File: Marlin/temperature.h [hunk 2]
```diff
@@ -50,6 +50,10 @@
   #define EXTRUDER_IDX  active_extruder
 #endif
 
+#if ENABLED(PIDTEMP)
+extern int pid_method;
+#endif
+
```
Comment (Blue-Marlin) [inline Marlin/temperature.h:56]: Having this variable in 'temperature.h' - gives other .cpp files a hint about `pid_method` when temperature.h is included.
But when `pid_method` is declared 'static' there it is not accessible.
Up to now there is no need for access to `pid_method` from other parts of Marlin - so you could remove this entry here.
As soon as you make `pid_method` configurable in the lcd-menu you need it here - but then the 'static' above does not make sense.

**Comment by Blue-Marlin** [comment]
Does this work for you?
All __my__ experiments with alternate settings to 'Classic PID' resulted in worse behaviour of the PID.

How about offering the new parameter in the menu?
How about making a default configurable?
How about storing the actual configured value in EEPROM?

File: Marlin/temperature.h [hunk 3]
```diff
@@ -50,6 +50,10 @@
   #define EXTRUDER_IDX  active_extruder
 #endif
 
+#if ENABLED(PIDTEMP)
+extern int pid_method;
```
Comment (thinkyhead) [inline Marlin/temperature.h:54]: Two spaces indentation here.

File: Marlin/temperature.cpp (hunk 1 above)
Comment (thinkyhead) [reply Marlin/temperature.cpp:89]: Please make this a `uint8_t`.

File: Marlin/Marlin_main.cpp [hunk 4]
```diff
@@ -6778,6 +6778,7 @@ inline void gcode_M303() {
     bool u = code_seen('U') && code_value_bool();
 
     float temp = code_seen('S') ? code_value_temp_abs() : (e < 0 ? 70.0 : 150.0);
+    pid_method = code_seen('O') ? code_value_int() : 0;  // 0 = Classic PID, 1 = Some Overshoot, 2 = No Overshoot
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp:6781]: Add here:
```cpp
if (pid_method > 2) pid_method = 0;
```

**Pull request closed without merging**